From 77cf25cb3d866fe76b863e362495050c0fa49eb0 Mon Sep 17 00:00:00 2001
From: Kai Vehmanen <kai.vehmanen@linux.intel.com>
Date: Fri, 3 Jan 2020 06:36:29 +0530
Subject: [PATCH] FROMGIT: ASoC: SOF: align snd-hda-codec-hdmi support with upstream

Incremental patch to align v5.0 tree with the final version
of the "add support for snd-hda-codec-hdmi" patch series
in upstream.

Contains fixes between v7 and v8 versions to following
commits in upstream:

139c7febad1a ASoC: SOF: Intel: add support for snd-hda-codec-hdmi
0f163110256a ASoC: Intel: skl-hda-dsp-generic: fix include guard name
7de9a47c8971 ASoC: Intel: skl-hda-dsp-generic: use snd-hda-codec-hdmi

Signed-off-by: Kai Vehmanen <kai.vehmanen@linux.intel.com>
---
 sound/soc/intel/boards/skl_hda_dsp_common.c | 3 ---
 sound/soc/sof/intel/hda-codec.c             | 2 +-
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/sound/soc/intel/boards/skl_hda_dsp_common.c b/sound/soc/intel/boards/skl_hda_dsp_common.c
index 3479dbbb0f8b..eb419e1ec42b 100644
--- a/sound/soc/intel/boards/skl_hda_dsp_common.c
+++ b/sound/soc/intel/boards/skl_hda_dsp_common.c
@@ -143,9 +143,6 @@ int skl_hda_hdmi_jack_init(struct snd_soc_card *card)
 		return skl_hda_hdmi_build_controls(card);
 
 	list_for_each_entry(pcm, &ctx->hdmi_pcm_list, head) {
-		if (!pcm)
-			continue;
-
 		component = pcm->codec_dai->component;
 		snprintf(jack_name, sizeof(jack_name),
 			 "HDMI/DP, pcm=%d Jack", pcm->device);
diff --git a/sound/soc/sof/intel/hda-codec.c b/sound/soc/sof/intel/hda-codec.c
index d62f068b617b..73f53916867b 100644
--- a/sound/soc/sof/intel/hda-codec.c
+++ b/sound/soc/sof/intel/hda-codec.c
@@ -83,7 +83,7 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address)
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC)
 	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
 	struct hdac_hda_priv *hda_priv;
-	struct snd_soc_acpi_mach_params *mach_params = 0;
+	struct snd_soc_acpi_mach_params *mach_params = NULL;
 	struct snd_sof_pdata *pdata = sdev->pdata;
 #endif
 	struct hda_bus *hbus = sof_to_hbus(sdev);
-- 
2.17.1

